src/drivers/indexeddb.js

```diff
@@ -399,37 +468,44 @@ function iterate(iterator, callback) {
 
     var promise = new Promise(function(resolve, reject) {
         self.ready().then(function() {
-            var dbInfo = self._dbInfo;
-            var store = dbInfo.db.transaction(dbInfo.storeName, 'readonly')
-                .objectStore(dbInfo.storeName);
-
-            var req = store.openCursor();
-            var iterationNumber = 1;
-
-            req.onsuccess = function() {
-                var cursor = req.result;
-
-                if (cursor) {
-                    var value = cursor.value;
-                    if (_isEncodedBlob(value)) {
-                        value = _decodeBlob(value);
-                    }
-                    var result = iterator(value, cursor.key,
-                        iterationNumber++);
-
-                    if (result !== void(0)) {
-                        resolve(result);
-                    } else {
-                        cursor.continue();
-                    }
-                } else {
-                    resolve();
+            createTransaction(self._dbInfo, 'readonly', function(err, transaction) {
+                if (err) {
+                    return reject(err);
                 }
-            };
 
-            req.onerror = function() {
-                reject(req.error);
-            };
+                try {
+                    var store = transaction.objectStore(self._dbInfo.storeName);
+                    var req = store.openCursor();
+                    var iterationNumber = 1;
+
+                    req.onsuccess = function() {
+                        var cursor = req.result;
+
+                        if (cursor) {
+                            var value = cursor.value;
+                            if (_isEncodedBlob(value)) {
+                                value = _decodeBlob(value);
+                            }
+                            var result = iterator(value, cursor.key,
+                                iterationNumber++);
+
+                            if (result !== void(0)) {
```

Comment:
To be clear, I just changed the indentation here, but this is an opportunity to add some comments as well :)